server/src/daemon/service_manager.rs

```diff
@@ -0,0 +1,24 @@
+use std::collections::HashMap;
+
+use zbus::zvariant::{ObjectPath, OwnedObjectPath};
+
+use super::session::Session;
+
+#[derive(Debug, Default)]
+pub struct ServiceManager {
+    sessions: HashMap<OwnedObjectPath, Session>,
+}
+
+impl ServiceManager {
+    pub fn session(&self, path: ObjectPath<'_>) -> Option<Session> {
```

Comment:
yes, as i said above, take a ref here and if you need an OwnedObjectPath, take one. Or take a generic param & avoid calling into() in the codebase.